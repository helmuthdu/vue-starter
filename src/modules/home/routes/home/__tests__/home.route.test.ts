import { mount, VueWrapper } from '@vue/test-utils';
import HomeRoute from '../home.route.vue';

describe('Route -> Home', () => {
  let wrapper: VueWrapper<any>;

  beforeEach(() => {
    wrapper = mount(HomeRoute, {
      global: {
        mocks: { $t: (t: string) => t }
      }
    });
  });

  it('should renders with props', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
