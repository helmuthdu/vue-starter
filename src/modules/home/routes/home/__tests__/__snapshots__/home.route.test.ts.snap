// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route -> Home should renders with props 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__emits": Object {},
      "__props": Array [
        Object {},
        Array [],
      ],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
      id="app"
    >
      <div
        class="columns"
      >
        <div
          class="column is-8-desktop is-offset-2-desktop"
        >
          <div
            class="content"
          >
            <h3>
              What's included
            </h3>
            <p>
              The 
              <code>
                npm
              </code>
               dependencies included in 
              <code>
                package.json
              </code>
               are:
            </p>
            <ul>
              <li>
                <code>
                  <router-link
                    to="[object Object]"
                  >
                    COMMON.ABOUT
                  </router-link>
                </code>
              </li>
              <li>
                <code>
                  <a
                    href="https://github.com/jgthms/bulma"
                  >
                    bulma
                  </a>
                </code>
              </li>
              <li>
                <code>
                  <a
                    href="https://github.com/sass/node-sass"
                  >
                    node-sass
                  </a>
                </code>
                 to compile your own Sass file 
              </li>
              <li>
                <code>
                  <a
                    href="https://github.com/postcss/postcss-cli"
                  >
                    postcss-cli
                  </a>
                </code>
                 and 
                <code>
                  <a
                    href="https://github.com/postcss/autoprefixer"
                  >
                    autoprefixer
                  </a>
                </code>
                 to add support for older browsers 
              </li>
              <li>
                <code>
                  <a
                    href="https://babeljs.io/docs/usage/cli/"
                  >
                    babel-cli
                  </a>
                </code>
                , 
                <code>
                  <a
                    href="https://github.com/babel/babel-preset-env"
                  >
                    babel-preset-env
                  </a>
                </code>
                 and 
                <code>
                  <a
                    href="https://github.com/jmcriffey/babel-preset-es2015-ie"
                  >
                    babel-preset-es2015-ie
                  </a>
                </code>
                 for compiling ES6 JavaScript files 
              </li>
            </ul>
            <blockquote
              class="blockquote"
            >
               “First, solve the problem. Then, write the code.” 
              <footer>
                <small>
                  <em>
                    —John Johnson
                  </em>
                </small>
              </footer>
            </blockquote>
          </div>
          <table
            class="table is-bordered is-fullwidth"
          >
            <thead>
              <tr>
                <th>
                  Type
                </th>
                <th>
                  CSS class
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <th>
                  Columns
                </th>
                <td>
                  <code>
                    <a
                      href="http://bulma.io/documentation/columns/basics"
                    >
                      columns
                    </a>
                  </code>
                  <code>
                    <a
                      href="http://bulma.io/documentation/columns/basics"
                    >
                      column
                    </a>
                  </code>
                </td>
              </tr>
              <tr>
                <th>
                  Layout
                </th>
                <td>
                  <code>
                    <a
                      href="http://bulma.io/documentation/layout/section"
                    >
                      section
                    </a>
                  </code>
                  <code>
                    <a
                      href="http://bulma.io/documentation/layout/container"
                    >
                      container
                    </a>
                  </code>
                  <code>
                    <a
                      href="http://bulma.io/documentation/layout/footer"
                    >
                      footer
                    </a>
                  </code>
                </td>
              </tr>
              <tr>
                <th>
                  Elements
                </th>
                <td>
                  <code>
                    <a
                      href="http://bulma.io/documentation/elements/button"
                    >
                      button
                    </a>
                  </code>
                  <code>
                    <a
                      href="http://bulma.io/documentation/elements/content"
                    >
                      content
                    </a>
                  </code>
                  <code>
                    <a
                      href="http://bulma.io/documentation/elements/title"
                    >
                      title
                    </a>
                  </code>
                  <code>
                    <a
                      href="http://bulma.io/documentation/elements/title"
                    >
                      subtitle
                    </a>
                  </code>
                </td>
              </tr>
              <tr>
                <th>
                  Form
                </th>
                <td>
                  <code>
                    <a
                      href="http://bulma.io/documentation/form/general"
                    >
                      field
                    </a>
                  </code>
                  <code>
                    <a
                      href="http://bulma.io/documentation/form/general"
                    >
                      control
                    </a>
                  </code>
                </td>
              </tr>
              <tr>
                <th>
                  Helpers
                </th>
                <td>
                  <code>
                    <a
                      href="http://bulma.io/documentation/modifiers/typography-helpers/"
                    >
                      has-text-centered
                    </a>
                  </code>
                  <code>
                    <a
                      href="http://bulma.io/documentation/modifiers/typography-helpers/"
                    >
                      has-text-weight-semibold
                    </a>
                  </code>
                </td>
              </tr>
            </tbody>
          </table>
          <div
            class="content"
          >
            <p>
              If you want to 
              <strong>
                learn more
              </strong>
              , follow these links:
            </p>
          </div>
          <div
            class="field is-grouped"
          >
            <p
              class="control"
            >
              <a
                class="button is-medium is-primary"
                href="http://bulma.io"
              >
                <strong
                  class="has-text-weight-semibold"
                >
                  Bulma homepage
                </strong>
              </a>
            </p>
            <p
              class="control"
            >
              <a
                class="button is-medium is-link"
                href="http://bulma.io/documentation/overview/start/"
              >
                <strong
                  class="has-text-weight-semibold"
                >
                  Documentation
                </strong>
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {},
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {},
      "reload": [Function],
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.7",
  },
  "__isFunctionalComponent": false,
  "__setProps": [Function],
  "componentVM": Object {
    "$i18n": Object {},
    "$t": [Function],
    "__emitted": Object {},
  },
  "rootVM": Object {
    "$i18n": Object {},
    "$t": [Function],
    "__emitted": Object {},
  },
}
`;
